mm2src/coins/eth/web3_transport.rs

```diff
@@ -74,12 +78,21 @@ pub struct Web3Transport {
     event_handlers: Vec<RpcTransportEventHandlerShared>,
 }
 
+#[derive(Debug, Display)]
+pub enum Web3TransportError {
+    #[display(fmt = "{}: ", _0)]
+    InvalidUri(String),
+}
+
 impl Web3Transport {
     #[allow(dead_code)]
-    pub fn new(urls: Vec<String>) -> Result<Self, String> {
+    pub fn new(urls: Vec<String>) -> Web3TransportResult<Self> {
```

Comment:
Please consider returning `web3::error::Error` error instead of adding `Web3TransportError`.